promql/engine_test.go

```diff
@@ -739,8 +786,38 @@ load 10s
 			Interval: 5 * time.Second,
 		},
 		{
-			Query:      "rate(bigmetric[1s])",
-			MaxSamples: 1,
+			Query:        `sum(metric)`,
+			MaxSamples:   8,
+			TotalSamples: 9,
+			PeakSamples:  7,
+			Result: Result{
+				nil,
+				Matrix{Series{Metric: labels.Labels{}, Points: []Point{{T: 0, V: 1}, {T: 5000, V: 1}, {T: 10000, V: 2}}}},
+				nil,
+			},
+			Start:    time.Unix(0, 0),
+			End:      time.Unix(10, 0),
+			Interval: 5 * time.Second,
+		},
+		{
+			Query:        `count(metric)`,
```

Comment:
This duplicates the above test

Reply:
@brian-brazil any example for what you expect to be a complex subquery?

Reply:
Anything that's not just a selector, so a function or operator.